### ISSUE TITLE
[a11y]: combobox - clear button not accessible by keyboard

### ISSUE THREAD

**Issue by patrickobrien1** [carbon-design-system/carbon#19263]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.76.0

### React version

v16.14.0

### Automated testing tool and ruleset

N/A

### Description

When using keyboard input to move focus across the screen, the combobox component does not permit keyboard access to the "x" button that eliminate input inside the combobox. Without this being accessible via keyboard input, it prevents keyboard users from knowing how to eliminate text from within a combobox.

### WCAG 2.1 Violation

2.1.1 Keyboard

### Reproduction/example

https://stackblitz.com/edit/github-9gy9vrze?file=src%2Fmain.jsx

### Steps to reproduce

Create a combo box component on a page, insert some value into it, once the "x" appears on the right side of the combo box attempt to tab to give that button keyboard focus. You are prevented from accessing this button via the keyboard.

### Suggested Severity

Severity 4 = Unrelated to a user task, has a workaround or does not need a workaround.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by annawen1** [comment]
@patrickobrien1 Looking into this a bit deeper, this does not appear to be a bug as the `ComboBox` input is able to be cleared by Escape key. This is listed as a way to clear the combobox in the https://www.w3.org/WAI/ARIA/apg/patterns/combobox/ website
